src/java/org/apache/cassandra/utils/ByteBufferUtil.java

```diff
@@ -726,6 +726,53 @@ public static int getShortLength(ByteBuffer bb, int position)
         return getUnsignedShort(bb, position);
     }
 
+    /**
+     * An efficient way to write the type {@code bytes} of a long
+     *
+     * @param register - the long value to be written
+     * @param bytes - the number of bytes the register occupies. Valid values are between 1 and 8 inclusive.
+     * @throws IOException
+     */
+    public static void putBytes(ByteBuffer b, int off, long register, int bytes)
+    {
+        switch (bytes)
+        {
+            case 0:
+                break;
+            case 1:
+                b.put(off, (byte)(register >>> 56));
+                break;
+            case 2:
+                b.putShort(off, (short)(register >> 48));
+                break;
+            case 3:
+                b.putShort(off, (short)(register >> 48));
+                b.put(off + 2, (byte)(register >> 40));
+                break;
+            case 4:
+                b.putInt(off, (int)(register >> 32));
+                break;
+            case 5:
+                b.putInt(off, (int)(register >> 32));
+                b.put(off + 4, (byte)(register >> 24));
+                break;
+            case 6:
+                b.putInt(off, (int)(register >> 32));
+                b.putShort(off + 4, (short)(register >> 16));
+                break;
+            case 7:
+                b.putInt(off, (int)(register >> 32));
+                b.putShort(off + 4, (short)(register >> 16));
```

Comment:
Has the offset + 4 and 5 issue

Reply:
This should have been removed entirely, my mistake.